### PR TITLE
New provider for address/geocode: LocationIQ

### PR DESCRIPTION
**Pull request by jnv**
## Description

Add new provider LocationIQ for Geocode and ReverseGeocode use cases.

The data format is equivalent with Nominatim, so the only things I changed were endpoints and added security.

## Motivation and Context

This is per request from the provider. I itend to include them in the comparison article: https://superface.ai/blog/geocoding-apis-comparison-1

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: grid/address/geocoding/maps/locationiq.suma [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        street = input.streetAddress
+        city = input.addressLocality
+        state = input.addressRegion
+        country = input.addressCountry
+        postalcode = input.postalCode
+      }
+    }
+
+    response 200 "application/json" {
+      return map error if (statusCode !== 200 || !body.length) {
```
Comment (tmladek) [inline grid/address/geocoding/maps/locationiq.suma]: How can `statusCode !== 200` in a `response 200` block?

File: grid/address/geocoding/maps/locationiq.suma [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    request {
+      query {
+        format = 'json'
+        lat = input.latitude
+        lon = input.longitude
+      }
+    }
+
+    response 200 "application/json" {
+      return map error if (statusCode !== 200) {
```
Comment (tmladek) [inline grid/address/geocoding/maps/locationiq.suma]: (ditto)

File: grid/address/geocoding/maps/__snapshots__/locationiq.test.ts.snap [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+`;
+
+exports[`address/geocoding/locationiq ReverseGeocode when all inputs are correct should reverse geocode geographical coordinates 1`] = `
+Ok {
+  "value": Array [
+    Object {
+      "addressCountry": "us",
+      "addressLocality": "New York",
+      "addressRegion": "undefined, Kings County, New York",
```
Comment (tmladek) [inline grid/address/geocoding/maps/__snapshots__/locationiq.test.ts.snap]: Is "undefined" correct (not preventable) here?

**Review by tmladek — DISMISSED** [review]
lgtm, small confusions

File: grid/address/geocoding/maps/locationiq.suma (hunk 1 above)
Comment (jnv) [reply grid/address/geocoding/maps/locationiq.suma]: Weird, right? I just copied this from Nominatim map.

File: grid/address/geocoding/maps/__snapshots__/locationiq.test.ts.snap (hunk 3 above)
Comment (jnv) [reply grid/address/geocoding/maps/__snapshots__/locationiq.test.ts.snap]: Right, that's wrong.

File: grid/address/geocoding/maps/locationiq.suma [hunk 4]
```diff
@@ -88,14 +88,15 @@ map ReverseGeocode {
...
-        mappedAddress.addressRegion = body.address.city_district;
-      }
+      addressRegion = body.address.city_district || ''
       set if (body.address.county) {
-        mappedAddress.addressRegion = `${mappedAddress.addressRegion}, ${body.address.county}`;
+        addressRegion = addressRegion ? `${addressRegion}, ${body.address.county}` : body.address.county;
       }
       set if (body.address.state) {
-        mappedAddress.addressRegion = `${mappedAddress.addressRegion}, ${body.address.state}`
+        addressRegion = addressRegion ? `${addressRegion}, ${body.address.state}` : body.address.state;
```
Comment (jnv) [inline grid/address/geocoding/maps/locationiq.suma]: BTW, I think this could be more elegant by using an array, but let's keep it now.

**Review by janhalama — APPROVED** [review]
LGTM

**Pull request merged**